docs/proposals/diagnose-experience.md

```diff
@@ -0,0 +1,247 @@
+  - Example: `DEBUG: [HTTPResponse] Status: 201 Created, Headers: {Location: /v2/oras-demo/blobs/uploads/uuid}, Body: {}`
+
+### 3. **Log Before and After Critical Operations**
+- **Operation Logs:** Log before performing critical operations and after completing them, including success or failure status.
+  - Example: `DEBUG: Starting upload of layer 2 of 3 for repository oras-demo`
+  - Example: `DEBUG: Successfully uploaded layer 2 of 3 for repository oras-demo`
+
+- **State Logs:** Log important state information before and after key operations or decisions.
+  - Example: `DEBUG: Current retry attempt: 1, Max retries: 3`
+
+### 4. **Error and Exception Handling**
+- **Catch and Log Exceptions:** Always catch exceptions and log them with relevant context and stack traces.
+  - Example: `ERROR: Exception occurred in fetchManifest: Network timeout while accessing /v2/oras-demo/manifests/latest`
+  
+- **Error Codes:** Include specific error codes to facilitate quick identification and resolution.
+  - Example: `ERROR: [ErrorCode: 504] Network timeout while accessing /v2/oras-demo/manifests/latest`
+
+### 5. **Environment and Configuration Details**
+- **Environment Information:** Log details about the environment where the tool is running, such as OS version and architecture, tool version, and environment variables (excluding sensitive data).
+  - Example: `DEBUG: Running on OS: Ubuntu 20.04, Arch: Linux AMD64 ORAS version: v1.2.0`
+
+- **Configuration Settings:** Include current configuration settings that might affect the execution or behavior of the tool.
+  - Example: `DEBUG: Configuration - Registry URL: https://myregistry.io, Timeout: 30s, RetryCount: 3`
+
+### 5. **Include Actionable Information**
+- **Guidance:** Where possible, provide suggestions within logs about potential fixes or next steps for resolving issues.
+  - Example: `DEBUG: [Action] Check network connectivity and retry the operation. Error occurred while accessing /v2/oras-demo/manifests/latest`
+
+- **Diagnostic Tips:** Include information that can assist in diagnosing issues like configuration settings, environment variables, or system states.
+  - Example: `DEBUG: Current registry URL: https://myregistry.io, Timeout setting:
+
+### 6. **Avoid Logging Sensitive Information**
+- **Privacy:** Abstain from logging sensitive information such as passwords, personal data, or security tokens.
+  - Example: `DEBUG: Attempting to authenticate user [UserID: usr123]` (without password details)
+
+- **Compliance:** Ensure that logs adhere to relevant data protection and privacy regulations.
+  - Example: Anonymize or obfuscate sensitive customer data in logs.
+
+## Proposals for ORAS CLI
+
+- Deprecate the global flag `--debug` and only remain `--verbose` to avoid ambiguous usage. 
+- Make the debug logs as an optional output controlled by `--verbose` via a parameter. Debug logs should be sent to stderr. 
```

Comment:
So, we are saying that we will still have two parameters? Is this correct? Or will the command parameter be `--verbose more-verbose`? This is a bit confusing how will it work.

Reply:
Are there going to be levels of debug/verbosity? It seems to me we are talking more about debug output rather than verbosity.

Reply:
Recommend introducing instead of any flag and the environment variable is totally not supported. 
```
export ORAS_DEBUG=1
oras pull ...
```

Reply:
Per [community meeting discussion on Sep 3](https://hackmd.io/P-O6n222TcSMoJgHmTTduw?view#Notes), ORAS maintainers agreed to not introduce an additional environment variable as a global switch to enable debug logs since `--debug` is intuitive enough.